iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/DroopCurveAdderImpl.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2025, SuperGrid Institute (https://www.supergrid-institute.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.network.impl;
+
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.impl.DroopCurveImpl.SegmentImpl;
+
+import java.util.*;
+
+/**
+ * @author Denis Bonnand {@literal <denis.bonnand at supergrid-institute.com>}
+ */
+class DroopCurveAdderImpl<O extends Validable & AcDcConverter<?>> implements DroopCurveAdder {
+
+    private final O owner;
+
+    private final HashSet<DroopCurve.Segment> segments = new HashSet<>();
```

Comment:
Same here, the hashset is useless, as it is required by the adder to create segment in the right order and with continuity (why not!), you can just use a List